### PR TITLE
code review, vertical layout fix for mobile phones

### PR DESCRIPTION
**Pull request by marykravets**

### CODE REVIEW

File: app/src/main/java/com/maxcraft/stackcalculator/Calculator.java [hunk 1]
```diff
@@ -6,9 +6,9 @@
 /**
  * Created by Maksym.Kravtsov on 1/23/2016.
  */
-public class Calculator
+class Calculator
```
Comment (maxcraft) [inline app/src/main/java/com/maxcraft/stackcalculator/Calculator.java]: The class is intended to be public and independent.

File: app/src/main/java/com/maxcraft/stackcalculator/Calculator.java [hunk 2]
```diff
@@ -6,9 +6,9 @@
 /**
  * Created by Maksym.Kravtsov on 1/23/2016.
  */
-public class Calculator
+class Calculator
 {
-	private Stack< Double > mStack = new Stack<>();
+	private final Stack< Double > mStack = new Stack<>();
```
Comment (maxcraft) [inline app/src/main/java/com/maxcraft/stackcalculator/Calculator.java]: Not sure if that's a good idea

File: app/src/main/java/com/maxcraft/stackcalculator/ListInverter.java [hunk 3]
```diff
@@ -10,7 +10,7 @@
 /**
  * Created by Maksym.Kravtsov on 1/23/2016.
  */
-public class ListInverter< E > implements List< E >
+class ListInverter< E > implements List< E >
```
Comment (maxcraft) [inline app/src/main/java/com/maxcraft/stackcalculator/ListInverter.java]: This class is intended to be public and independent.

File: app/src/main/java/com/maxcraft/stackcalculator/ListInverter.java [hunk 4]
```diff
@@ -32,13 +32,13 @@ public boolean add( E object )
 	}
 
 	@Override
-	public boolean addAll( int location, Collection< ? extends E > collection )
+	public boolean addAll(int location, @NonNull Collection< ? extends E > collection )
```
Comment (maxcraft) [inline app/src/main/java/com/maxcraft/stackcalculator/ListInverter.java]: nonNull adds just compile-time checking. Also put back my space character.

File: app/src/main/java/com/maxcraft/stackcalculator/ListInverter.java [hunk 5]
```diff
@@ -32,13 +32,13 @@ public boolean add( E object )
...
 	@Override
-	public boolean addAll( int location, Collection< ? extends E > collection )
+	public boolean addAll(int location, @NonNull Collection< ? extends E > collection )
 	{
 		return mList.addAll( invertLocation( location ), collection );
 	}
 
 	@Override
-	public boolean addAll( Collection< ? extends E > collection )
+	public boolean addAll(@NonNull Collection< ? extends E > collection )
```
Comment (maxcraft) [inline app/src/main/java/com/maxcraft/stackcalculator/ListInverter.java]: nonNull adds just compile-time checking. Also put back my space character.

File: app/src/main/java/com/maxcraft/stackcalculator/ListInverter.java [hunk 6]
```diff
@@ -56,7 +56,7 @@ public boolean contains( Object object )
 	}
 
 	@Override
-	public boolean containsAll( Collection< ? > collection )
+	public boolean containsAll(@NonNull Collection< ? > collection )
```
Comment (maxcraft) [inline app/src/main/java/com/maxcraft/stackcalculator/ListInverter.java]: nonNull adds just compile-time checking. Also put back my space character.

File: app/src/main/java/com/maxcraft/stackcalculator/ListInverter.java [hunk 7]
```diff
@@ -118,13 +116,13 @@ public boolean remove( Object object )
 	}
 
 	@Override
-	public boolean removeAll( Collection< ? > collection )
+	public boolean removeAll(@NonNull Collection< ? > collection )
```
Comment (maxcraft) [inline app/src/main/java/com/maxcraft/stackcalculator/ListInverter.java]: nonNull adds just compile-time checking. Also put back my space character.

File: app/src/main/java/com/maxcraft/stackcalculator/ListInverter.java [hunk 8]
```diff
@@ -118,13 +116,13 @@ public boolean remove( Object object )
...
 	@Override
-	public boolean removeAll( Collection< ? > collection )
+	public boolean removeAll(@NonNull Collection< ? > collection )
 	{
 		return mList.removeAll( collection );
 	}
 
 	@Override
-	public boolean retainAll( Collection< ? > collection )
+	public boolean retainAll(@NonNull Collection< ? > collection )
```
Comment (maxcraft) [inline app/src/main/java/com/maxcraft/stackcalculator/ListInverter.java]: nonNull adds just compile-time checking. Also put back my space character.

File: app/src/main/java/com/maxcraft/stackcalculator/ListInverter.java [hunk 9]
```diff
@@ -157,9 +155,9 @@ public int size()
 
 	@NonNull
 	@Override
-	public < T > T[] toArray( T[] array )
+	public < T > T[] toArray(@NonNull T[] array )
```
Comment (maxcraft) [inline app/src/main/java/com/maxcraft/stackcalculator/ListInverter.java]: nonNull adds just compile-time checking. Also put back my space character.

File: app/src/main/java/com/maxcraft/stackcalculator/MainActivity.java [hunk 10]
```diff
@@ -16,25 +16,23 @@
...
+	private final HashMap< View, Runnable > mBtns = new HashMap<>();
 	private View.OnClickListener mBtnClickListener;
 
 	private Calculator mCalculator;
 	private TextView mIndicatorTextView;
-	private ListView mStackListView;
 	private ArrayAdapter< Double > mAdapter;
-	private StringBuffer mInputString = new StringBuffer();
-	private NumberFormat mNumFormat = new DecimalFormat( "0.##############" );
+	private final StringBuffer mInputString = new StringBuffer();
```
Comment (maxcraft) [inline app/src/main/java/com/maxcraft/stackcalculator/MainActivity.java]: final will make in initialized fewer times I actually need that.

File: app/src/main/java/com/maxcraft/stackcalculator/MainActivity.java [hunk 11]
```diff
@@ -16,25 +16,23 @@
...
+	private final NumberFormat mNumFormat = new DecimalFormat( "0.##############" );
```
Comment (maxcraft) [inline app/src/main/java/com/maxcraft/stackcalculator/MainActivity.java]: final will make in initialized fewer times I actually need that.

File: app/src/main/java/com/maxcraft/stackcalculator/MainActivity.java [hunk 12]
```diff
@@ -16,25 +16,23 @@
...
 
 	@Override
 	protected void onCreate( Bundle savedInstanceState )
 	{
 		super.onCreate( savedInstanceState );
 		setContentView( R.layout.activity_main );
-		mDecimalFormatSymbols = new DecimalFormatSymbols();
+		DecimalFormatSymbols mDecimalFormatSymbols = new DecimalFormatSymbols();
```
Comment (maxcraft) [inline app/src/main/java/com/maxcraft/stackcalculator/MainActivity.java]: Remove 'm' prefix if you moved the field to a variable.

File: app/src/main/java/com/maxcraft/stackcalculator/MainActivity.java [hunk 13]
```diff
@@ -59,9 +57,9 @@ public Object onRetainCustomNonConfigurationInstance()
 
 	private void initViews()
 	{
-		mIndicatorTextView = ( TextView )findViewById( R.id.indicator_txt );
-		mStackListView = ( ListView )findViewById( android.R.id.list );
-		mAdapter = new ArrayAdapter< Double >( this, android.R.layout.simple_list_item_1, new ListInverter<>( mCalculator.getStack() ) );
+		mIndicatorTextView = findViewById( R.id.indicator_txt );
+		ListView mStackListView = findViewById(android.R.id.list);
```
Comment (maxcraft) [inline app/src/main/java/com/maxcraft/stackcalculator/MainActivity.java]: Please remove 'm' prefix for local variables.

File: app/src/main/java/com/maxcraft/stackcalculator/MainActivity.java [hunk 14]
```diff
@@ -59,9 +57,9 @@ public Object onRetainCustomNonConfigurationInstance()
 
 	private void initViews()
 	{
-		mIndicatorTextView = ( TextView )findViewById( R.id.indicator_txt );
-		mStackListView = ( ListView )findViewById( android.R.id.list );
-		mAdapter = new ArrayAdapter< Double >( this, android.R.layout.simple_list_item_1, new ListInverter<>( mCalculator.getStack() ) );
+		mIndicatorTextView = findViewById( R.id.indicator_txt );
+		ListView mStackListView = findViewById(android.R.id.list);
+		mAdapter = new ArrayAdapter<>(this, android.R.layout.simple_list_item_1, new ListInverter<>(mCalculator.getStack()));
```
Comment (maxcraft) [inline app/src/main/java/com/maxcraft/stackcalculator/MainActivity.java]: Please return my code formatting.

**Review by maxcraft — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/maxcraft/stackcalculator/ListInverter.java [hunk 15]
```diff
@@ -32,7 +32,7 @@ public boolean add( E object )
 	}
 
 	@Override
-	public boolean addAll( int location, Collection< ? extends E > collection )
+	public boolean addAll(int location, Collection< ? extends E > collection )
```
Comment (maxcraft) [inline app/src/main/java/com/maxcraft/stackcalculator/ListInverter.java]: Still missing space character.

File: app/src/main/java/com/maxcraft/stackcalculator/ListInverter.java [hunk 16]
```diff
@@ -159,7 +157,7 @@ public int size()
 	@Override
 	public < T > T[] toArray( T[] array )
 	{
-		return mList.toArray( array );
+		return array;
```
Comment (maxcraft) [inline app/src/main/java/com/maxcraft/stackcalculator/ListInverter.java]: Are you sure that will work this way?

**Review by maxcraft — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/maxcraft/stackcalculator/ListInverter.java (hunk 16 above)
Comment (marykravets) [reply app/src/main/java/com/maxcraft/stackcalculator/ListInverter.java]: yes.

**Review by maxcraft — APPROVED** [review]

**Pull request merged**